blog-posts/cloudfront-cache-efficiency.md

```diff
@@ -0,0 +1,154 @@
+---
+path: "/blog/cloudfront-cache-efficiency/"
+date: "2022-02-02"
+title: "Improve cache efficiency between CloudFront and the browser"
+featuredImage: images/cloudfront-cache-efficiency/cover.jpg 
+attribution:
+    creator: Faris Mohammed 
+    source: https://unsplash.com/photos/d30sszrW7Vw
+author: Fabian Dietenberger 
+authorSummary: "Senior Developer at Satellytes"
+seoMetaText: Learn how to improve Caching Headers in CloudFront to reduce traffic and increase page speed.
+teaserText: Learn how to improve Caching Headers in CloudFront to reduce traffic and increase page speed.
+---
+
+Lately we were trying out CloudFront to improve our API response and caching times. We wanted to cache the response
+in the browser for 10 seconds, and on CloudFront for 60 seconds. 
+
+To do this we set the `Cache-Control` header in our origin like this: `Cache-Control: max-age=10, s-maxage=60`
+
+This `Cache-Control` header has 2 directives: `max-age` and `s-maxage`. `max-age` defines the caching time in seconds. 
+In our case this means, the browser should cache the resource for 10 seconds. After 10 seconds, the browser should 
+request the resource and cache it for 10 seconds again.  
+The `s-maxage` header is used to set the caching time of the resource in our CDN in seconds (if not set,`max-age` is used).
+With the first request, the CDN caches the resource for 60 seconds. After that time, the cache gets deleted and the resource
+needs to be fetched by CloudFront from the origin again.
+
+To test this header, we created a small sample application within AWS Lambda and CloudFront as CDN. The Lambda returns
+the given caching header and a small JSON body. Let’s check within the browsers network console what happens when we do
+multiple requests within 70 seconds:
+
+![Requests are only partially cached](images/cloudfront-cache-efficiency/1-problem.png)
+
+What do we see here?
+
+1. The first request hit’s the Lambda function. This is takes around 1 second, as the function is cold and uncached by
+   CloudFront. The response get’s now cached by the browser and CloudFront.
+2. The next 3 requests are not send, as they are within 10 seconds, which is the `max-age`. The browser just returns the
+   cache. The request time is therefore 0 seconds.
+3. The next 7 requests are all done the next 50 seconds. They hit the CloudFront cache, the response time is therefore
+   around 20ms.
+4. 60 seconds after the first requests, CloudFront clears the cache. The request hits the lambda
+   function, so it takes around 700ms. After that requests, the same procedure starts again.
+
+What is wrong here?
+
+1. The browser should cache every request for 10 seconds. But it only caches the first requests for the `max-age` and #
```

Comment:
Something is wrong with the formatting here
<img width="883" alt="Bildschirmfoto 2022-02-07 um 08 20 30" src="https://user-images.githubusercontent.com/57712895/152742615-ec1ea903-4928-405d-b9fc-b5b13ef233a1.png">

Reply:
Reformatting the markdown in IntelliJ sometimes breaks it. Thanks for figuring it out :)